### ISSUE TITLE
Internal error in the S-expression parser, please report upstream.

### ISSUE THREAD

**Issue by jberdine** [ocaml/dune#1548]
I am seeing this error trying to build https://github.com/ocaml-ppx/ocamlformat/tree/master with dune master. As far as I can tell by bisecting, the issue was introduced by #1517 (I can build ocamlformat with 3d3e0775 but not 57575d80).

I'm not seeing how to find more information on what sexp dune is unhappy with.

**Comment by diml** [comment]
Thanks for bisecting the error. I couldn't reproduce the it, does the error message mentions a specific file? If yes, could you send the file?

**Comment by jberdine** [comment]
Sorry for the delay, but I think that I have understood this now. It seems that dune is trying to interpret the dune binary in a local opam switch as a dune file:
```
File "_opam/bin/dune", line 1, characters 0-0:
Error: Internal error in the S-expression parser, please report upstream.
```
If I add `(ignored_subdirs (_opam))` to the project's toplevel `dune` file, then sanity is restored.

**Comment by ejgallego** [comment]
Oh so #1560 is a duplicate of this issue.

**Comment by ejgallego** [comment]
This should be fixed in 1.6 , do you mind checking @jberdine ?

**Comment by jberdine** [comment]
Yes, works here with 1.6, thanks!
